plugins/auto-sizes/includes/improve-calculate-sizes.php

```diff
@@ -135,50 +138,124 @@ function auto_sizes_filter_image_tag( $content, array $parsed_block, WP_Block $b
 /**
  * Modifies the sizes attribute of an image based on layout context.
  *
- * @param int    $id           The image id.
- * @param string $size         The image size data.
- * @param string $align        The image alignment.
- * @param string $resize_width Resize image width.
+ * @since n.e.x.t
+ *
+ * @param int    $id                   The image id.
+ * @param string $size                 The image size data.
+ * @param string $align                The image alignment.
+ * @param string $resize_width         Resize image width.
+ * @param bool   $is_parent_block      Check if image block has parent block.
+ * @param string $ancestor_block_align The ancestor block alignment.
  * @return string The sizes attribute value.
  */
-function auto_sizes_calculate_better_sizes( int $id, string $size, string $align, string $resize_width ): string {
-	$sizes = '';
+function auto_sizes_calculate_better_sizes( int $id, string $size, string $align, string $resize_width, bool $is_parent_block, string $ancestor_block_align ): string {
 	$image = wp_get_attachment_image_src( $id, $size );
 
 	if ( false === $image ) {
-		return $sizes;
+		return '';
 	}
 
 	// Retrieve width from the image tag itself.
 	$image_width = '' !== $resize_width ? (int) $resize_width : $image[1];
 
+	if ( $is_parent_block ) {
```

Comment:
I think a simpler way to do this is to compare the `$ancestor_block_align` value to the block's `align` attribute value, and use whichever is smaller in a single call to `auto_sizes_get_sizes_by_block_alignments()`.

Reply:
See https://github.com/WordPress/performance/pull/1704#discussion_r1861496634 it get issue for single image block.